Userland/Utilities/route.cpp

```diff
@@ -0,0 +1,198 @@
+        }
+    };
+
+    if (!modify_action) {
+        auto file = TRY(Core::File::open("/proc/net/route", Core::OpenMode::ReadOnly));
+        auto file_contents = file->read_all();
+        auto json = TRY(JsonValue::from_string(file_contents));
+
+        outln("Kernel IP routing table");
+
+        for (auto& column : columns)
+            print_column(column, column.title);
+        outln();
+
+        Vector<JsonValue> sorted_regions = json.as_array().values();
+        quick_sort(sorted_regions, [](auto& a, auto& b) {
+            return a.as_object().get("destination").to_string() < b.as_object().get("destination").to_string();
+        });
+
+        for (auto& value : sorted_regions) {
+            auto& if_object = value.as_object();
+
+            auto destination = if_object.get("destination").to_string();
+            auto gateway = if_object.get("gateway").to_string();
+            auto genmask = if_object.get("genmask").to_string();
+            auto interface = if_object.get("interface").to_string();
+
+            if (destination_column != -1)
+                columns[destination_column].buffer = destination;
+            if (gateway_column != -1)
+                columns[gateway_column].buffer = gateway;
+            if (genmask_column != -1)
+                columns[genmask_column].buffer = genmask;
+            if (interface_column != -1)
+                columns[interface_column].buffer = interface;
+
+            for (auto& column : columns)
+                print_column(column, column.buffer);
+            outln();
+        };
+    }
+
+    if (modify_action) {
+        const bool action_add = (strcmp(modify_action, "add") == 0);
+        const bool action_del = (strcmp(modify_action, "del") == 0);
+
+        if (!action_add && !action_del) {
+            warnln("Invalid modify action: {}", modify_action);
+            return 1;
+        }
+
+        if (!value_host_address && !value_network_address) {
+            warnln("No target host or network specified");
+            return 1;
+        }
+
+        Optional<IPv4Address> destination;
+
+        if (value_host_address)
+            destination = AK::IPv4Address::from_string(value_host_address);
+
+        if (value_network_address)
+            destination = AK::IPv4Address::from_string(value_network_address);
+
+        if (!destination.has_value()) {
+            warnln("Invalid destination IPv4 address");
+            return 1;
+        }
+
+        auto gateway = AK::IPv4Address::from_string(value_gateway_address);
+        if (!gateway.has_value()) {
+            warnln("Invalid gateway IPv4 address: '{}'", value_gateway_address);
+            return 1;
+        }
+
+        auto genmask = AK::IPv4Address::from_string(value_netmask_address);
+        if (!genmask.has_value()) {
+            warnln("Invalid genmask IPv4 address: '{}'", value_netmask_address);
+            return 1;
+        }
+
+        int fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_IP);
```

Comment:
Since you are already using `serenity_main` please use the `Core::System::socket()` version instead of the `LibC` version you are using here. Then you can use the `TRY(..)` macro to do error propagation for you automatically.